.github/workflows/genesis-verify.yml

```diff
@@ -0,0 +1,59 @@
+name: genesis-verify
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths:
+      - "misc/deployments/**/genesis.json"
+
+jobs:
+  verify:
+    strategy:
+      fail-fast: false
+      matrix:
+        testnet: ["test4.gno.land", "test5.gno.land"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v41
+        with:
+          files: "misc/deployments/${{ matrix.testnet }}/genesis.json"
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22"
+
+      - name: Install dependencies
+        run: go mod download
+
+      - name: Build gnogenesis
+        run: make -C contribs/gnogenesis
+
+      - name: Verify each genesis file
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            echo "Verifying $file"
+            gnogenesis verify -genesis-path $file
+          done
+
+      - name: Build gnoland
+        run: make -C gno.land install.gnoland
+
+      - name: Running latest gnoland with each genesis file
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            echo "Running gnoland with $file"
+            timeout 60s gnoland start -lazy=true --genesis $file || exit_code=$?
+            if [ $exit_code -eq 124 ]; then
+              echo "Gnoland run successfully for 60 seconds"
```

Comment:
This message should be `Gnoland genesis state generated successfully`

Reply:
83be3b2f0445fbb398afbf6537133325c11094b0